clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/ShareConsumerTest.java

```diff
@@ -2058,7 +2058,8 @@ public void testComplexShareConsumer() throws Exception {
         // all messages which can be read are read, some would be redelivered
         TestUtils.waitForCondition(complexCons1::isDone, 45_000L, () -> "did not close!");
 
-        assertTrue(prodState.count().get() < complexCons1.recordsRead());
+        assertTrue(prodState.count().get() <= complexCons1.recordsRead(),
```

Comment:
This assertion failed in your build :(

Reply:
I'll keep the condition as the same but add some info as part of failed message.